components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2263,6 +2308,19 @@ where
                             exist_peer
                         ));
                     }
+                    if exist_peer.get_is_witness() != peer.get_is_witness() {
```

Comment:
How about merging with above condition? Same store with different ID may not be duplicate anyway.